src/renderer.rs

```diff
@@ -146,7 +162,7 @@ where
 
         let max_children = old.children.len().max(new.children.len());
 
-        for index in 0..max_children {
+        for index in (0..max_children).rev() {
```

Comment:
Remove from last to first, since otherwise, if any of the children are being deleted, the index doesn't make sense.